packages/rest/src/http/metadata.ts

```diff
@@ -430,6 +443,12 @@ export function createMetadataInfo(program: Program, options?: MetadataInfoOptio
     return true;
   }
 
+  function isOptional(property: ModelProperty, visibility: Visibility): boolean {
+    // Item flag is not a real visibility.
+    visibility &= ~Visibility.Item;
```

Comment:
And sorry, I just read that merge-patch can only replace an entire array, it cannot patch array elements. I didn't know that, but it makes sense. And if so, Visbility.Update and only Visibility.Update actually seems correct. array element schemas should retain their non-optionality. So it seems like my proof-of-concept had this right, but only via pure luck. Sheesh, sorry.